FlowCrypt/Controllers/Setup/SetupGenerateKeyViewController.swift

```diff
@@ -144,25 +147,62 @@ private actor Service {
                            for: user.email)
 
         if storageMethod == .memory {
-            let passPhrase = PassPhrase(value: passPhrase, fingerprintsOfAssociatedKey: encryptedPrv.key.fingerprints)
+            let passPhrase = PassPhrase(
+                value: passPhrase,
+                fingerprintsOfAssociatedKey: encryptedPrv.key.fingerprints
+            )
             passPhraseService.savePassPhrase(with: passPhrase, storageMethod: .memory)
         }
 
-        let updateKey = attester.updateKey(
+        await updateKeyAndWelcome(
             email: userId.email,
-            pubkey: encryptedPrv.key.public,
-            token: storage.token
+            publicKey: encryptedPrv.key.public,
+            viewController: viewController
         )
+    }
 
-        try awaitPromise(await viewController.alertAndSkipOnRejection(
-            updateKey,
-            fail: "Failed to submit Public Key")
-        )
-        let testWelcome = attester.testWelcome(email: userId.email, pubkey: encryptedPrv.key.public)
-        try awaitPromise(await viewController.alertAndSkipOnRejection(
-            testWelcome,
-            fail: "Failed to send you welcome email")
-        )
+    private func updateKeyAndWelcome(
```

Comment:
This is too complicated. You can remove the error handling/alert code in this case, and simply add two attester calls into `setupAccount`

Reply:
So we don't notify user about error here?

Reply:
I had a second look. For `attester.testWelcome`, we don't have to notify user about errors (it's low importance - this sends user a welcome message, which is not crucial). Can just directly add in `setupAccount`:

```ts
// sending welcome email is not crucial, so we don't handle errors
_ = try? await attester.testWelcome(email: email, pubkey: publicKey)
```

and remove `testWelcome` method. Then let's rename `updateKeyAndWelcome` to `submitKeyToAttesterAndShowAlertOnFailure`, and remove `testWelcome` from that. That should clean things up nicely.